lib/tapioca/dsl/helpers/active_record_column_type_helper.rb

```diff
@@ -148,6 +147,23 @@ def serialized_column_type(column_type)
             "T.untyped"
           end
         end
+
+        sig { params(column_type: T.untyped, getter_type: String).returns(String) }
+        def maybe_nilable_type(column_type, getter_type)
```

Comment:
I don't love this method name, curious about any alternatives. For example, `as_nilable_column_type`?

Reply:
Another option could be define something like this:

```ruby
sig { params(column_type: T.untyped).returns(T::Boolean) }
def not_nilable_serialized_column?(column_type)
  column_type.is_a?(ActiveRecord::Type::Serialized) &&
    column_type.coder.is_a?(ActiveRecord::Coders::YAMLColumn) &&
    [Array.singleton_class, Hash.singleton_class].include?(column_type.coder.object_class.singleton_class)
end
```

And change `type_for`:

```ruby
if column&.null
  getter_type = as_nilable_type(getter_type) unless not_nilable_serialized_column?(column_type)
  return [getter_type, as_nilable_type(setter_type)]
end
```

What do you think?

Reply:
I've changed the implementation.